src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -109,12 +110,14 @@ def __init__(
         self.prompt_batch_threshold = prompt_batch_threshold
         self.force_max_tokens = force_max_tokens
 
+        self.engine_args["cache_bools"] = self.cache_bools
         self.engine = Pipeline.create_engine(
             self.onnx_file_path,
             self.engine_type,
             self.engine_args,
             self.context,
         )
+        self.kv_cache = KVCache({}, 0, [0])  # TODO: populate with correct values
```

Comment:
@markurtz @SageMoore what do we want for the initial kv cache? zeros of the correct shape? do we need param names?

Reply:
Very good question. Also, how will initial kv cache differ whether we are initially running multitoken inference or directly start with the singletoken inference?